src/libraries/System.Net.Sockets/src/System/Net/Sockets/TCPClient.cs

```diff
@@ -285,23 +285,41 @@ public void Connect(IPAddress[] ipAddresses, int port)
             if (NetEventSource.IsEnabled) NetEventSource.Exit(this);
         }
 
-        public Task ConnectAsync(IPAddress address, int port) =>
-            Task.Factory.FromAsync(
-                (targetAddess, targetPort, callback, state) => ((TcpClient)state!).BeginConnect(targetAddess, targetPort, callback, state),
-                asyncResult => ((TcpClient)asyncResult.AsyncState!).EndConnect(asyncResult),
-                address, port, state: this);
-
-        public Task ConnectAsync(string host, int port) =>
-            Task.Factory.FromAsync(
-                (targetHost, targetPort, callback, state) => ((TcpClient)state!).BeginConnect(targetHost, targetPort, callback, state),
-                asyncResult => ((TcpClient)asyncResult.AsyncState!).EndConnect(asyncResult),
-                host, port, state: this);
-
-        public Task ConnectAsync(IPAddress[] addresses, int port) =>
-            Task.Factory.FromAsync(
-                (targetAddresses, targetPort, callback, state) => ((TcpClient)state!).BeginConnect(targetAddresses, targetPort, callback, state),
-                asyncResult => ((TcpClient)asyncResult.AsyncState!).EndConnect(asyncResult),
-                addresses, port, state: this);
+        public Task ConnectAsync(IPAddress address, int port)
+        {
+            if (NetEventSource.IsEnabled) NetEventSource.Enter(this, address);
+
+            Task result = CompleteConnectAsync(Client.ConnectAsync(address, port));
+
+            if (NetEventSource.IsEnabled) NetEventSource.Exit(this);
```

Comment:
Are you intending `Exit` to be called prior to the connect being completed?

Reply:
Yeah, I'm just mirroring what the BeginConnect methods are doing below.  I'd also be perfectly happy removing all of these Enter/Exits if folks are ok with that.

Reply:
>I'd also be perfectly happy removing all of these Enter/Exits if folks are ok with that.

Are you proposing removing the tracing completely? Or just refactoring it differently? 

I would prefer we don't remove tracing.

Reply:
> Are you proposing removing the tracing completely? Or just refactoring it differently?

I'm simply not clear on the benefits of these Enter/Exit traces that we have in some (not all) methods.

Reply:
I've been doing a lot of thinking about tracing w.r.t. to the reverse-proxy work.

These particular traces were blindly added as we ported lots of .NET Framework code to .NET Core. At the time we wanted to replicate the debug/diagnostic tracing we had with TraceListener and the debug-build GlobalLog of networking code in .NET Framework.

I agree we might want to rethink this but I would prefer we do that holistically as part of a separate PR/effort.

Reply:
> I agree we might want to rethink this but I would prefer we do that holistically as part of a separate PR/effort.

Fine by me.  Sounds like you're on top of it.  Thanks.